storage.go

```diff
@@ -474,6 +474,9 @@ func (s *Storage) read(ctx context.Context, path string, w io.Writer, opt pairSt
 		Bucket: aws.String(s.name),
 		Key:    aws.String(rp),
 	}
+	if opt.HasOffset && opt.HasSize {
```

Comment:
What if the user only inputs `size` or `offset`? (NOTE: they are both valid.)

- size=x, offset=0 -> read from 0 to x
- size=0, offset=x -> read from x to the end
- size=x, offset=y -> read from x to x+y

We need to handle all of them.

Reply:
OK